src/wasm/wasm-binary.cpp

```diff
@@ -2240,7 +2240,8 @@ void WasmBinaryBuilder::readTypes() {
               "The only allowed trivial supertype for functions is func");
           }
         } else {
-          if (basicSuper != HeapType::data) {
+          // "struct" used to be "data".
+          if (basicSuper != HeapType::struct_) {
             throwError("The only allowed trivial supertype for structs and "
                        "arrays is data");
```

Comment:
No, this is parsing the nonstandard "struct_subtype" and "array_subtype" formats that explicitly use `data` as the placeholder supertype. Since `struct` now has the encoding that `data` used to have, that's what we expect here. No module will ever be produced that uses `array` here. I'll expand the comment.